Yoast/yoastseo-amp
# Composer: remove CodeClimate test reporter dev dependency

## Summary
This PR can be summarized in the following changelog entry:
* _N/A_

## Relevant technical choices:

I'm suggesting to remove the CodeClimate test reporter dependency for the following reasons:
1. The CodeClimate test reporter repository is marked as deprecated.
    Ref: https://github.com/codeclimate/php-test-reporter/blob/master/README.md
2. The repository as-is is using abandoned dependencies, such a `guzzle/guzzle` (now `guzzlehttp/guzzle`).
3. But most of all, because the _test reporter_ is only useful if there are actual tests and the YoastSEO AMP plugin currently **_does not have any_**.


## Test instructions

This PR can be tested by following these steps:
* _N/A_ As this dependency wasn't being used in the first place, not having it should make no difference at all.
